plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxMetadata.java

```diff
@@ -0,0 +1,240 @@
+import static io.trino.plugin.influxdb.InfluxProjectedManager.buildProjection;
+import static java.util.Objects.requireNonNull;
+
+public class InfluxMetadata
+        implements ConnectorMetadata
+{
+    private final InfluxClient client;
+
+    @Inject
+    public InfluxMetadata(InfluxClient client)
+    {
+        this.client = requireNonNull(client, "client is null");
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession session)
+    {
+        return ImmutableList.copyOf(client.getSchemaNames());
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> optionalSchemaName)
+    {
+        Set<String> schemaNames = optionalSchemaName.map(ImmutableSet::of)
+                .orElseGet(() -> ImmutableSet.copyOf(client.getSchemaNames()));
+
+        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
+        for (String schemaName : schemaNames) {
+            for (String tableName : client.getTableNames(schemaName)) {
+                builder.add(new SchemaTableName(schemaName, tableName));
+            }
+        }
+        return builder.build();
+    }
+
+    @Override
+    public InfluxTableHandle getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        if (!listSchemaNames(session).contains(schemaTableName.getSchemaName())) {
+            return null;
+        }
+
+        Optional<InfluxTable> table = client.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        if (table.isEmpty()) {
+            return null;
+        }
+
+        return new InfluxTableHandle(schemaTableName.getSchemaName(), schemaTableName.getTableName(), TupleDomain.all());
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return getTableMetadata(((InfluxTableHandle) tableHandle).toSchemaTableName());
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        InfluxTableHandle influxTableHandle = (InfluxTableHandle) tableHandle;
+
+        Optional<InfluxTable> table = client.getTable(influxTableHandle.getSchemaName(), influxTableHandle.getTableName());
+        if (table.isEmpty()) {
+            throw new TableNotFoundException(influxTableHandle.toSchemaTableName());
+        }
+
+        ImmutableMap.Builder<String, ColumnHandle> columnHandles = ImmutableMap.builder();
+        for (InfluxColumn column : table.get().getColumns()) {
+            columnHandles.put(column.getName(), new InfluxColumnHandle(column.getName(), toTrinoType(column.getType()).orElse(null)));
+        }
+        return columnHandles.buildOrThrow();
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
```

Comment:
oh, sorry about that. Because time column is so special and important to time-series db (that's why "time" is a mondatory key in every point, and for influxQL ,whatever you select "time" or not, it always return the time value in result additionally),  and in most scenarios the time value is necessary. so i left it  not a hidden column.  and for this reason, i override those method in BaseConnectorTest, they all affected by the time column